test/transcoding/OpAllAny.ll

```diff
@@ -3,54 +3,104 @@
 ; RUN: FileCheck < %t.txt %s --check-prefix=CHECK-SPIRV
 ; RUN: llvm-spirv %t.bc -o %t.spv
 ; RUN: spirv-val %t.spv
+; RUN: llvm-spirv -r %t.spv -o %t.rev.spv.bc --spirv-target-env=SPV-IR
+; RUN: llvm-dis < %t.rev.spv.bc | FileCheck %s --check-prefix=CHECK-SPV-IR
 ; RUN: llvm-spirv -r %t.spv -o %t.rev.bc
 ; RUN: llvm-dis < %t.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM
-; RUN: llvm-spirv -r %t.spv -o %t.rev.bc --spirv-target-env=SPV-IR
-; RUN: llvm-dis < %t.rev.bc | FileCheck %s --check-prefix=CHECK-SPV-LLVM
 
-; CHECK-LLVM: call spir_func i32 @_Z3allDv2_i(
-; CHECK-LLVM: call spir_func i32 @_Z3anyDv2_i(
-
-; CHECK-SPV-LLVM: call spir_func i32 @_Z11__spirv_AllDv2_i(
-; CHECK-SPV-LLVM: call spir_func i32 @_Z11__spirv_AnyDv2_i(
-
-; CHECK-SPV-LLVM: declare spir_func i32 @_Z11__spirv_AllDv2_i(<2 x i32>)
-; CHECK-SPV-LLVM: declare spir_func i32 @_Z11__spirv_AnyDv2_i(<2 x i32>)
+; This test checks SYCL relational builtin any and all with vector input types.
 
 ; CHECK-SPIRV: 2 TypeBool [[BoolTypeID:[0-9]+]]
-; CHECK-SPIRV: 4 All [[BoolTypeID]]
+
+; CHECK-SPIRV: 4 Any [[BoolTypeID]]
 ; CHECK-SPIRV: 4 Any [[BoolTypeID]]
+; CHECK-SPIRV: 4 Any [[BoolTypeID]]
+; CHECK-SPIRV: 4 Any [[BoolTypeID]]
+; CHECK-SPIRV: 4 All [[BoolTypeID]]
+; CHECK-SPIRV: 4 All [[BoolTypeID]]
+; CHECK-SPIRV: 4 All [[BoolTypeID]]
+; CHECK-SPIRV: 4 All [[BoolTypeID]]
+
+; CHECK-SPV-IR: call spir_func i1 @_Z11__spirv_AnyDv2_b(<2 x i1>
+; CHECK-SPV-IR: call spir_func i1 @_Z11__spirv_AnyDv2_b(<2 x i1>
+; CHECK-SPV-IR: call spir_func i1 @_Z11__spirv_AnyDv2_b(<2 x i1>
+; CHECK-SPV-IR: call spir_func i1 @_Z11__spirv_AnyDv2_b(<2 x i1>
+; CHECK-SPV-IR: call spir_func i1 @_Z11__spirv_AllDv2_b(<2 x i1>
+; CHECK-SPV-IR: call spir_func i1 @_Z11__spirv_AllDv2_b(<2 x i1>
+; CHECK-SPV-IR: call spir_func i1 @_Z11__spirv_AllDv2_b(<2 x i1>
+; CHECK-SPV-IR: call spir_func i1 @_Z11__spirv_AllDv2_b(<2 x i1>
+
+; CHECK-LLVM: call spir_func i32 @_Z3anyDv2_i(<2 x i32>
+; CHECK-LLVM: call spir_func i32 @_Z3anyDv2_i(<2 x i32>
+; CHECK-LLVM: call spir_func i32 @_Z3anyDv2_i(<2 x i32>
+; CHECK-LLVM: call spir_func i32 @_Z3anyDv2_i(<2 x i32>
+; CHECK-LLVM: call spir_func i32 @_Z3allDv2_i(<2 x i32>
+; CHECK-LLVM: call spir_func i32 @_Z3allDv2_i(<2 x i32>
+; CHECK-LLVM: call spir_func i32 @_Z3allDv2_i(<2 x i32>
+; CHECK-LLVM: call spir_func i32 @_Z3allDv2_i(<2 x i32>
```

Comment:
Maybe the test is a little confusing.

Sycl header in https://github.com/intel/llvm won't generate <2 x i64> input for any/all. Sycl header already trunc i64 input to i8 because spirv builtin only accepts i8 input, see https://github.com/intel/llvm/blob/cabb43fdb137f6cc65fc6486a1b6be1300185df7/clang/lib/Sema/SPIRVBuiltins.td#L856

spirv-translator itself won't generate <2 x i64> input neither, because it follows spirv spec.

Are there any other source that will generate <2 x i64>? I'm not aware of that. If such source may exist, we can use i64 return type in SPIRVToOCLBase::visitCallSPIRVAnyAll, what's your opinion?

Reply:
OpenCL C specification:
```
int all(igentype x)
The argument type igentype refers to the built-in signed integer types i.e. char, charn, short, shortn, int, intn, long and longn.
```
Clang can generate <2 x i64>: https://godbolt.org/z/rbYz1xnKf

Reply:
If the input IR is <2 x i64> any/all from OpenCL, spvtoocl20 doesn't do transformation and IR is not modified.
spvtoocl20 translates spirv friendly IR, so its input type is defined by spirv friendly IR format and available implementations such as in sycl header or llvm-spirv translator itself. 

Since this patch doesn't change the behavior of using i32 as representation type for OpenCL output, I'd suggest to open a new issue for possible i64 input. How do you think?

Reply:
Actually, you're right. Since SPIRV Friendly IR representation operates on boolean inputs then extending them to i32 vs extending them to i64 doesn't make any difference. Your code is correct.

I'm just wondering why OpenCL C allows for different vector types `char, charn, short, shortn, int, intn, long and longn.`, while SPIRV representation operates only on boolean vectors.

Reply:
yes, such difference exists, and SYCL builtin signatures are mostly aligned with OpenCL C builtin .

Reply:
I'm not aware of the rational either.

Reply:
OpenCL C representation allows for more vector types, but it actually care only for one bit anyway (`"Returns 1 if the most significant bit...is set"`), so it looks like SPIRV representation was just simplified, but it still represents the same functionallity.

Reply:
> I'm just wondering why OpenCL C allows for different vector types char, charn, short, shortn, int, intn, long and longn., while SPIRV representation operates only on boolean vectors.

Simply because SPIR-V is not limited to represent OpenCL C (SPIR-V has more ties to Vulkan than to OpenCL).